src/toil/jobStores/fileJobStore.py

```diff
@@ -193,6 +193,9 @@ def _importFile(self, otherCls, url, sharedFileName=None):
                 fd, absPath = self._getTempFile()  # use this to get a valid path to write to in job store
                 os.close(fd)
                 os.unlink(absPath)
+                # remove the .tmp extension and add the original file name
+                (noExt,ext) = os.path.splitext(absPath)
+                absPath = noExt + '-' + os.path.basename(url.path)
```

Comment:
This only fixes files that enter the jobStore from `importFile()`. The underlying filename is set from [`fileJobStore._getTempFile()`](https://github.com/BD2KGenomics/toil/blob/5819ff372f08f146ff17e7053bfe34f1347624a8/src/toil/jobStores/fileJobStore.py#L445)

Which is used to create the file name any time a file enters the jobStore, be it from importFile, writeGlobalFile, or writeGlobalFileStream. The fix should happen there instead.

Reply:
To add to this, it may be worth while to add a test that ensures that this behaves as expected and that the file still has the correct name (or at least extension) if it was brought into the jobStore from importFile, writeGlobalFile, or writeGlobalFileStream.